url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Agent.java

```diff
@@ -0,0 +1,105 @@
+package org.agetac.common.model.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.sign.AbstractModel;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Agent extends AbstractModel {
+
+	public enum Aptitude {
+	    COS,
+	    CA,
+	    CDG,
+	    CDC,
+	    AGENT
+	}
+	
+	private Aptitude aptitude;
+	private List<Agent> subordonnes;
+
+	public Agent() {
+		super(null, null, null);
+		this.aptitude = null;
+		this.subordonnes = null;
+	}
+
+	public Agent(String uid, String name, Aptitude aptitude, List<Agent> subordonnes) {
+		super(uid, name, null);
+		this.aptitude = aptitude;
+		this.subordonnes = subordonnes;
+	}
+
+	public Agent(JSONObject json) throws InvalidJSONException {
+		super(json);
+		
+		try{
+			this.aptitude = Aptitude.valueOf(json.getString("aptitude"));
+			JSONArray array = json.getJSONArray("subordonnes");
+			
+			this.subordonnes = new ArrayList<Agent>();
+			for (int i=0; i<array.length(); i++) {
+				this.subordonnes.add(new Agent(array.getJSONObject(i)));
+			}
+		}catch (JSONException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)